### ISSUE TITLE
We should combine forces!

### ISSUE THREAD

**Issue by karai17** [davisdude/mlib#12]
CPML and mlib have very similar goals, a lot of the same functionality, but we're both missing things from each other. I propose that we [combine forces](https://www.youtube.com/watch?v=VOTlJRcNS_g) to make CPML the ultimate general purpose Lua math library!

Firstly, I suggest we use CPML as a base since it is already written in a manner that you want mlib to evolve to (metatables, modularization, etc.) as discussed in issue #9.

Secondly, we're *really* attached to the name "Cirno's Perfect Math Library". It's a really funny joke for [those in the know](https://www.youtube.com/watch?v=5wFDWP5JwSM).

Thirdly, much of CPML is already documented using LDoc, although this still needs work.

Both libraries are fighting for the same goal and "competing", however informally, seems less useful than actually building one awesome tool together.

**Comment by davisdude** [comment]
Yeah, I agree. Object-based especially seems to be a lot more game-friendly. 

CPML also already has error-checking, as well as a few other things that I was planning on adding. It seems kind of stupid to work on them both independently when it could be better if we combined them. 

One side note though is that I'm pretty infrequent about when I have time to work on the library. But either way, I agree that it would be best to work together.

**Comment by karai17** [comment]
![](http://static2.fjcdn.com/comments/Anonymous+rolled+a+random+image+posted+in+comment+113+at+_0153cc9fad4caae8fd1cac56461ea43a.jpg)

COMBINE!

In all seriousness, Time itself isn't all that important. CPML already has a lot going for it so it's a matter of porting some of your code over into new modules, or expending other modules. Improving documentation, etc. You've been added to the CPML repo. Please try to be consistent with our code style ;) <3

**Comment by davisdude** [comment]
Alright. LDoc seems like the way to go anyway-- working on documentation for hours at a time isn't exactly what I call fun. 

I'll try and look at your coding style and port some stuff over when I can

**Comment by karai17** [comment]
Cool! our coding style is pretty simple. underscore_naming, tab to indent, space to align, align grouped = signs, etc.

**Comment by davisdude** [comment]
Alright, I have some style questions: 
- When should the version be incremented (semantic versioning, I'm assuming)?
- Is there any reason you use double quotes for the last part of `local current_folder = (...):gsub('%.[^%.]+$', '') .. "."`?

**Comment by karai17** [comment]
We are really loose on the version number, it basically means nothing right
now  but we should probably consider actually incrementing it properly at
some point.

That final "." is there for a reason. current_folder is only used for
requiring sister modules so it just means we can type:

local vec3 = require(current_folder .. "vec3")

Instead of:

local vec3 = require(current_folder .. ".vec3")

**Comment by davisdude** [comment]
I know that, I was wondering why you use single quotes everywhere else, and then use double quotes there.

**Comment by karai17** [comment]
Ah! We generally use double quotes but I think we copied the main section
of that line from somewhere else. There is no practical reason, feel free
to change them all to double quotes if you like.

**Comment by davisdude** [comment]
Oh, okay. Thanks

**Comment by buckle2000** [comment]
How is it now?

**Comment by davisdude** [comment]
I haven't really contributed a lot to CPML yet. Right now I'm pretty busy, although I'm also slowly re-working through the library, as there were some small bugs.
